lang/cs/Org.Apache.REEF.Network/NetworkService/WritableNsMessage.cs

```diff
@@ -0,0 +1,184 @@
+        {
+            _sourceIdString = reader.ReadString();
+            _destIdString = reader.ReadString();
+            int messageCount = reader.ReadInt32();
+
+            Data = new List<T>();
+
+            for (int index = 0; index < messageCount; index++)
+            {
+                Data.Add(Activator.CreateInstance<T>());
+
+                if (Data[index] == null)
+                {
+                    throw new Exception("T type instance cannot be created from the stream data in Network Service Message");
+                }
+
+                Data[index].Read(reader);
+            }
+        }
+
+        /// <summary>
+        /// Writes the class fields.
+        /// </summary>
+        /// <param name="writer">The writer to which to write</param>
+        public void Write(IDataWriter writer)
+        {
+            writer.WriteString(SourceId.ToString());
+            writer.WriteString(DestId.ToString());
+            writer.WriteInt32(Data.Count);
+
+            foreach (var data in Data)
+            {
+                data.Write(writer);
+            }
+        }
+
+        /// <summary>
+        /// Read the class fields.
+        /// </summary>
+        /// <param name="reader">The reader from which to read </param>
+        /// <param name="token">The cancellation token</param>
+        public async Task ReadAsync(IDataReader reader, CancellationToken token)
+        {
+            _sourceIdString = await reader.ReadStringAsync(token);
+            _destIdString = await reader.ReadStringAsync(token);
+            int messageCount = await reader.ReadInt32Async(token);
+
+            Data = new List<T>();
+
+            for (int index = 0; index < messageCount; index++)
+            {
+                Data.Add(Activator.CreateInstance<T>());
+
+                if (Data[index] == null)
+                {
+                    throw new Exception("T type instance cannot be created from the stream data in Network Service Message");
+                }
+
+                await Data[index].ReadAsync(reader, token);
+            }
+        }
+
+        /// <summary>
+        /// Writes the class fields.
+        /// </summary>
+        /// <param name="writer">The writer to which to write</param>
+        /// <param name="token">The cancellation token</param>
+        public async Task WriteAsync(IDataWriter writer, CancellationToken token)
+        {
+            await writer.WriteStringAsync(SourceId.ToString(), token);
+            await writer.WriteStringAsync(DestId.ToString(), token);
+            await writer.WriteInt32Async(Data.Count, token);
+
+            foreach (var data in Data)
+            {
+                data.Write(writer);
+            }
+        }
+
+        /// <summary>
+        /// Converts Source and Destination strings to identifiers
+        /// </summary>
+        /// <param name="factory">Identifier factory</param>
+        public void ConvertStringToIdentifier(IIdentifierFactory factory)
```

Comment:
ohh u r saying this will happen automatically...If yes then yes this helps a lot :) ...I have created another jira 320 for including this in Wake and will send pull request soon.